### ISSUE TITLE
[KIEKER-1097] Live Demo Issues when using AspectJ

### ISSUE THREAD

**Issue by rju** [kieker-monitoring/kieker#2570]
JIRA Issue: [KIEKER-1097] Live Demo Issues when using AspectJ
Original Reporter: Nils Christian Ehmke
 *** 
<p>Follow up from KIEK-1242: When replacing SpringAOP-based instrumentation with AspectJ, the live demo is no longer available.</p>

[KIEKER-1097]: https://kieker-monitoring.atlassian.net/browse/KIEKER-1097?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by rju** [comment]
author nils-christian -- Wed, 15 Oct 2014 14:45:19 +0200
<p>First problem: When using AspectJ the JMX writer is <br class="atl-forced-newline" />- according to the log <br class="atl-forced-newline" />- not used. This has to be forced via the parameter <tt>-Dkieker.monitoring.configuration=webapps\jpetstore\WEB-INF\classes\META-INF\kieker.monitoring.properties</tt> in the ini file.</p>

<p>Second problem: The aop configuration is not loaded automatically, as it seems. This has to be forced via the parameter <tt>-Dkieker.monitoring.skipDefaultAOPConfiguration=true</tt> in the ini file.</p>

<p>There are still some errors during the weaving process though. Excluding the service layer within JPetStore (<tt>&lt;exclude within="org.mybatis.jpetstore.service..*"/&gt;</tt>) solves this problem, but the live demo is still not working. It seems like the Kieker monitoring is not even started although the classes are woven with the correct aspects.</p>

**Comment by rju** [comment]
author nils-christian -- Wed, 15 Oct 2014 15:10:00 +0200
<p>I correct myself about Kieker. Monitoring logs are written to the file system using the default monitoring.properties file.</p>

<p>The problem with the non-working live demo seems to be a class loading issue or perhaps classpath poisoning. Removing <tt>commons-logging-1.1.1.jar</tt> from the live demo's lib folder is the solution. However, I cannot explain exactly what happens during the weaving that leads to the problem in the first place...</p>

**Comment by rju** [comment]
author André van Hoorn -- Wed, 15 Oct 2014 15:50:42 +0200
<p>Replying to <span class="error">&#91;nie|comment:1&#93;</span>:<br/>
&gt; <del>Second problem: The aop configuration is not loaded automatically, as it seems. This has to be forced via the parameter <tt>-Dkieker.monitoring.skipDefaultAOPConfiguration=true</tt> in the ini file.</del><br/>
&gt; </p>

<p>(Re-)added parameter to <tt>start.ini</tt> (changeset:9ff868af4aa64591ec7f424842feb92b4aa80801/kieker-git)</p>

**Comment by rju** [comment]
author nils-christian -- Sun, 15 Mar 2015 08:57:55 +0100
<p>It seems that the monitoring leads to issues with the classloading of some logger classes (maybe because the livedemo uses the Kieker logging as well). I forced another logger implementation adding a parameter to the start.ini. The above exception still occurs, but both JPetStore and the LiveDemo seem to work. </p>

<p>I added the modifications to the master branch. If required, it can be cherrypicked for the 1.11 branch.</p>

<p>Corresponding changeset: 651cf204e0ba826bd1b41a98264e2ccad4ec0288</p>
